libs/core/langchain_core/language_models/chat_models.py

```diff
@@ -254,7 +265,14 @@ def stream(
                         generation = chunk
                     else:
                         generation += chunk
+
+                    if chunk.message.id != last_chunk_id:
+                        seen_chunk_ids.append(chunk.message.id)
+                        last_chunk_id = chunk.message.id
                 assert generation is not None
+            except self.exceptions_to_handle:
+                for chunk_id in seen_chunk_ids:
+                    yield RemoveMessageChunk(id=chunk_id)  # type: ignore[call-arg]
```

Comment:
i think we'd still want to raise an error at the end of this? otherwise the user code has now way of knowing that nothing has actually streamed

Reply:
yes, agreed